src/main/java/com/redhat/devtools/lsp4ij/LanguageServersRegistry.java

```diff
@@ -190,22 +192,32 @@ private void updateLanguages() {
         // the language received in InlayHintProviders is plain/text, we add it to support
         // LSP inlayHint, color for a file which is not linked to a language.
         distinctLanguages.add(PlainTextLanguage.INSTANCE);
-
-        // register LSPInlayHintsProvider + LSPColorProvider automatically
+        // register LSPInlayHintsProvider automatically
+        // for all languages associated with a language server.
+        updateDeclarativeInlayHintsProviders(distinctLanguages);
+        // register LSPColorProvider automatically
         // for all languages associated with a language server.
         updateInlayHintsProviders(distinctLanguages);
         // register LSPFindUsagesProvider automatically
         // for all languages associated with a language server.
         updateFindUsagesProvider(distinctLanguages);
     }
 
-    private void updateInlayHintsProviders(Set<Language> distinctLanguages) {
+    private void updateDeclarativeInlayHintsProviders(Set<Language> distinctLanguages) {
         LSPInlayHintsProvider lspInlayHintsProvider = new LSPInlayHintsProvider();
+        inlayHintsProviders.clear();
+        for (Language language : distinctLanguages) {
+            List<InlayProviderInfo> hints = new ArrayList<>();
+            hints.add(new InlayProviderInfo(lspInlayHintsProvider, LSPInlayHintsProvider.PROVIDER_ID, Collections.emptySet(), true, "Language Server"));
```

Comment:
Use LanguageServerBundle.message('....) to avoid hard coding "Language Server" string